hls-plugin-api/src/Ide/Types.hs

```diff
@@ -170,6 +170,8 @@ instance Show (IdeCommand st) where show _ = "<ide command>"
 data PluginDescriptor (ideState :: *) =
   PluginDescriptor { pluginId           :: !PluginId
                    -- ^ Unique identifier of the plugin.
+                   , pluginConfig       :: !PluginConfig
```

Comment:
This is a good question - I completely forgot about the custom config stuff. It probably makes sense to consolidate all the plugin config in a single place, will check.

Reply:
It's a field of `PluginConfig`, so in principle you could set the defaults using this mechanism instead... but the other one is nice because it's typed, I guess. I don't really know how this stuff works properly.

Reply:
Me neither, but it seems well thought out...